internal/services/cdn/cdn_frontdoor_route_disable_link_to_default_domain_resource.go

```diff
@@ -163,143 +164,173 @@ func resourceCdnFrontDoorRouteDisableLinkToDefaultDomainRead(d *pluginsdk.Resour
+			}
 
-	props := routeResp.RouteProperties
-	if props == nil {
-		return fmt.Errorf("%s properties are 'nil': %+v", *routeId, err)
-	}
+			// Since this unlink default domain resource always set the value to false
+			updateProps.LinkToDefaultDomain = cdn.LinkToDefaultDomainDisabled
 
-	resourceCustomDomains := d.Get("cdn_frontdoor_custom_domain_ids").([]interface{})
-	routeCustomDomains := flattenCdnFrontdoorRouteActivatedResourceArray(props.CustomDomains)
+			// NOTE: You must pull the Cache Configuration from the existing route else you will get a diff, because a nil value means disabled
+			if props.CacheConfiguration != nil {
+				updateProps.CacheConfiguration = props.CacheConfiguration
+			}
 
-	// validate the custom domains...
-	if err := validateCustomDomanLinkToDefaultDomainState(resourceCustomDomains, routeCustomDomains, routeId.RouteName, routeId.ProfileName); err != nil {
-		return err
-	}
+			updateParams := azuresdkhacks.RouteUpdateParameters{
+				RouteUpdatePropertiesParameters: &updateProps,
+			}
+
+			future, err := workaroundsClient.Update(routeCtx, routeId.ResourceGroup, routeId.ProfileName, routeId.AfdEndpointName, routeId.RouteName, updateParams)
+			if err != nil {
+				return fmt.Errorf("updating %s: %+v", id, err)
+			}
+			if err = future.WaitForCompletionRef(routeCtx, routeClient.Client); err != nil {
+				return fmt.Errorf("waiting for the update of %s: %+v", id, err)
+			}
+		}
 
-	d.Set("cdn_frontdoor_custom_domain_ids", resourceCustomDomains)
+		d.Set("cdn_frontdoor_route_id", routeId.ID())
+		d.Set("cdn_frontdoor_custom_domain_ids", customDomains)
+	}
 
-	return nil
+	return resourceCdnFrontDoorRouteDisableLinkToDefaultDomainRead(d, meta)
 }
 
 func resourceCdnFrontDoorRouteDisableLinkToDefaultDomainDelete(d *pluginsdk.ResourceData, meta interface{}) error {
-	routeClient := meta.(*clients.Client).Cdn.FrontDoorRoutesClient
-	workaroundsClient := azuresdkhacks.NewCdnFrontDoorRoutesWorkaroundClient(routeClient)
-	routeCtx, routeCancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
-	defer routeCancel()
+	client := meta.(*clients.Client).Cdn.FrontDoorRoutesClient
+	workaroundsClient := azuresdkhacks.NewCdnFrontDoorRoutesWorkaroundClient(client)
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
 
 	id, err := parse.FrontDoorRouteDisableLinkToDefaultDomainID(d.Id())
 	if err != nil {
 		return err
 	}
 
-	currentRoute := d.Get("cdn_frontdoor_route_id").(string)
-
-	// If this delete was due to a change you need to revert
-	// the old route not the new route...
-	if d.HasChange("cdn_frontdoor_route_id") {
-		if oldRoute, _ := d.GetChange("cdn_frontdoor_route_id"); oldRoute.(string) != "" {
-			currentRoute = oldRoute.(string)
-		}
-	}
+	oldRoute, _ := d.GetChange("cdn_frontdoor_route_id")
 
-	routeId, err := parse.FrontDoorRouteID(currentRoute)
+	route, err := parse.FrontDoorRouteIDInsensitively(oldRoute.(string))
```

Comment:
this is coming from the config which is validated and so should be being parsed case-sensitively:

```suggestion
	route, err := parse.FrontDoorRouteID(oldRoute.(string))
```

Reply:
Fixed.